test/e2e/CustomComponents.test.js

```diff
@@ -0,0 +1,68 @@
+const { execute } = require("../../lib");
+const { createMockStreams, allArgs } = require("./E2ETests");
+const lolex = require("lolex");
+const brs = require("brs");
+const { RoSGNode, BrsString, Callable, ValueKind } = brs.types;
+const path = require("path");
+
+class Foo extends RoSGNode {
+    constructor(initializedFields = [], name = "Foo") {
+        super([], name);
+        this.name = name;
+
+        this.registerDefaultFields(this.defaultFields);
+        this.registerInitializedFields(initializedFields);
+
+        this.sayHello = new Callable("sayHello", {
+            signature: {
+                args: [],
+                returns: ValueKind.String,
+            },
+            impl: (_interpreter) => {
+                return new BrsString("Foo Says Hello world");
+            },
+        });
+
+        this.registerMethods({
+            ifHelloWorld: [this.sayHello],
+        });
+    }
+}
+
+describe("Extending custom components", () => {
+    let outputStreams;
+    let clock;
+
+    beforeAll(() => {
+        clock = lolex.install({ now: 1547072370937 });
```

Comment:
I suspect we don't actually need to mock out the clock for these tests, since none of this appears to depend on the current time. Would you mind removing this?